### PR TITLE
Adding class for SQLite and crud functions

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Fixes #597

Changes proposed in this PR:

- Adding SQLite for the nonce database. Only runs when a valid Typesense URL is not provided.
- Testing the nonce database when no Typesense URL is provided.

### CODE REVIEW

**Comment by bogdanfazakas** [comment]
Hi @jamiehewitt15 are like going to create smaller PR's to integrate SQLite in the nodes flow ?

**Comment by jamiehewitt15** [comment]
> Hi @jamiehewitt15 are like going to create smaller PR's to integrate SQLite in the nodes flow ?

I'm still testing it locally, but I thought I would add it to this PR. I have marked it as draft for now.

File: src/components/database/index.ts [hunk 1]
```diff
@@ -608,34 +612,57 @@ export class DdoDatabase {
...
+          })
+          await this.provider.collections(this.schema.name).retrieve()
+        } catch (error) {
+          if (error instanceof TypesenseError && error.httpStatus === 404) {
+            await (this.provider as Typesense).collections().create(this.schema)
+          }
         }
+      } else {
+        // Fall back to SQLite
+        DATABASE_LOGGER.logMessageWithEmoji(
```
Comment (mariacarmina) [inline src/components/database/index.ts:636]: I think also @bogdanfazakas works on this type of integration to add elasticsearch, maybe we can have a factory class which retrun Typesense DB, ES DB, SQLITE.

**Review by mariacarmina — COMMENTED** [review]
a suggestion from my side.

**Review by mariacarmina — APPROVED** [review]
lgtm

**Review by alexcos20 — APPROVED** [review]

**Pull request merged**